pkg/providers/tinkerbell/hardware/csv.go

```diff
@@ -87,3 +90,24 @@ func ensureRequiredColumnsInCSV(unmatched []string) error {
 
 	return nil
 }
+
+// BuildHardwareYaml builds a hardware yaml from the csv at the provided path.
+func BuildHardwareYaml(path string) ([]byte, error) {
```

Comment:
```suggestion
// BuildHardwareYAML builds a hardware yaml from the csv at the provided path.
func BuildHardwareYAML(path string) ([]byte, error) {
```

Reply:
From what I understand from investigating, the `CreationTimestamp` in the `metav1.ObjectMeta` is optional and it's tagged with `omitempty`, but because it isn't a pointer it will never satisfy the empty check. There's an issue open but it seems to be difficult to do:

https://github.com/kubernetes/kubernetes/issues/67610

As for apiGroup, there is no `omitempty` tag on it [here](https://pkg.go.dev/k8s.io/api/core/v1#TypedLocalObjectReference), and I don't think we use it

Reply:
Yea that's why we have a separate ObjectMeta object here: https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/object_meta.go#L8

Are we saying we don't want to use this instead to simply get rid of the field as this is a patch approach? Since we use this for others it feels like we are doing a lot of extra work for hardware, but then again it could be the right way. Either way, we probably want to create an issue to be in sync with the others.

Reply:
I think this is fine for now but marshalling into JSON shouldn't result in nul fields output and I think that's probably something that needs addressing upstream.